.projenrc.js

```diff
@@ -105,3 +115,78 @@ project.eslint.addRules({
 });
 
 project.synth();
+
+function codeGenFetchAssemblies() {
+  const scriptPath = "scripts/fetch-assemblies.js";
+
+  const packagesPath = "public/packages";
+
+  const script = new SourceCode(project, scriptPath);
+  script.line(`// ${FileBase.PROJEN_MARKER}`);
+  script.line(fetchAssembly.toString());
+  script.line(`${fetchAssembly.name}('${packagesPath}')`);
+
+  project.gitignore.exclude(packagesPath);
+
+  const task = project.addTask("dev:fetch-assemblies");
+  task.exec(`node ${scriptPath}`);
+}
+
+/**
+ * Fetch an assembly by name and version, accepted via env variable for easier invocation.
+ * This fetches all the necessary assemblies into the packages path, which can later be used to instantiate
+ * a jsii type system for the specific package.
+ */
+function fetchAssembly(packagesPath) {
+  const packageName = process.env.PACKAGE_NAME;
+  const packageVersion = process.env.PACKAGE_VERSION;
+
+  if (!packageName) {
+    throw new Error(
+      `Package name missing. Please provide via 'PACKAGE_NAME' env variable.`
+    );
+  }
+
+  if (!packageVersion) {
+    throw new Error(
+      `Package version missing. Please provide via 'PACKAGE_VERSION' env variable`
+    );
+  }
+
+  const exec = require("child_process").exec;
+  const fs = require("fs");
+  const os = require("os");
+  const path = require("path");
+
+  if (packageVersion.startsWith("^")) {
+    packageVersion = packageVersion.substring(1, packageVersion.length);
+  }
+
+  const output = `${packagesPath}/${packageName}@${packageVersion}/jsii.json`;
+
+  const tempDir = fs.mkdtempSync(`${os.tmpdir()}/assembly`);
+  const assemblyPath = `${__dirname}/../${output}`;
+  const package = `${packageName}@${packageVersion}`;
+
+  if (fs.existsSync(assemblyPath)) {
+    console.log(`Assembly for ${package} already exists locally`);
+    return;
+  }
+
+  console.log(`Fetching assembly for ${package}`);
+  fs.mkdirSync(path.dirname(assemblyPath), { recursive: true });
+  exec(
+    `cd ${tempDir} && npm v ${packageName}@${packageVersion} dist.tarball | xargs curl | tar -xz`,
```

Comment:
`dist.tarball` is a special string that gives you the tarball URL from `npm view`.

Reply:
Nitpick: I kinda sorta would rather we `npm view` than `npm v`. Not everyone reading this is familiar with all `npm` commands, and abbreviations are mostly meant for when you are typing that in an interactive shell (IMHO).